src/rard/research/views/concordance.py

```diff
@@ -82,75 +253,165 @@ class ConcordanceCreateView(
     check_lock_object = "top_level_object"
 
     # create a concordance for an original text
-    model = Concordance
+    model = ConcordanceModel
     permission_required = ("research.add_concordance",)
-    fields = ("source", "identifier")
+    form_class = ConcordanceModelCreateForm
 
     def dispatch(self, request, *args, **kwargs):
         # need to ensure we have the lock object view attribute
         # initialised in dispatch
         self.top_level_object = self.get_original_text().owner
-        if not isinstance(self.get_original_text().owner, Fragment) and not isinstance(
-            self.get_original_text().owner, AnonymousFragment
-        ):
-            raise Http404
 
         return super().dispatch(request, *args, **kwargs)
 
+    def get(self, request, *args, **kwargs):
+        edition = self.kwargs.get("e_pk", None)
+        part_format = self.kwargs.get("p_pk", None)
+        concordance_form = ConcordanceModelCreateForm(
+            edition=edition, part_format=part_format
+        )
+        return render(
+            request,
+            "research/partials/concordance_form_section.html",
+            context=self.get_context_data(
+                concordance_form=concordance_form,
+                ediiton=edition,
+                part_format=part_format,
+            ),
+        )
+
+    def post(self, request, *args, **kwargs):
+        edition = request.POST.get("edition", self.kwargs.get("e_pk"))
+        new_identifier = request.POST.get("new_identifier", None)
+        identifier = request.POST.get("identifier", None)
+        display_order = request.POST.get("display_order", None)
+        if (
+            new_identifier is not None
+            and new_identifier.startswith("[")
+            and new_identifier.endswith("]")
+        ):
+            new_identifier = new_identifier[1:-1]
+
+        concordance_form = ConcordanceModelCreateForm(request.POST)
+        if concordance_form.is_valid():
+            if new_identifier:
+                new_identifier_instance = PartIdentifier.objects.create(
+                    edition=Edition.objects.get(pk=edition),
+                    value=concordance_form.cleaned_data["new_identifier"],
+                    display_order=concordance_form.cleaned_data["display_order"],
+                )
+                new_identifier_instance.save()
+                identifier = new_identifier_instance.pk
+            elif identifier is None:
+                identifier = get_part_format(edition).pk
+
+            if display_order is None:
```

Comment:
Did you check this works? I think it isn't working for a couple of reasons: 
* line 290 where you get display_order from the POST data actually returns an empty string rather than None, so this if block never gets called - empty strings evaluate to false though, so you could just use `if not display_order:`
* the other thing is that you're not actually doing anything with display_order here after updating the value.

I'd usually suggest using a Form's clean method to handle this sort of situation (see the Django docs on [form validation](https://docs.djangoproject.com/en/5.1/ref/forms/validation/#cleaning-and-validating-fields-that-depend-on-each-other)). When you call is_valid() on the concordance form above, it should run the clean method on the form and update the value.